### DISCUSSION TITLE
Subtracting background spectra from sample spectra

### DISCUSSION THREAD

**Discussion by phoebe-sch** [spectrochempy/spectrochempy#731] [Q&A] [answered]
Hi -
For my spectra, I have the sample spectra and the background spectra of the substrate the sample is on. I need to subtract the background spectra from the sample spectra before I begin identifying peaks. Is this doable with spectrochempy? I have searched the user's guide and haven't found background removal. If there is, can you direct me to it?

Thanks so much!

**Comment by atravert** [comment, marked as answer]
Hello,

Yes it is perefectly doable using the subtraction. For instance, if your sample spectra and background have been recorded in distinct files, you could do:

```python
import spectrochempy as scp

A = scp.read("mydata/sample_spectra.spg")
B = scp.read("mydata/background_spectrum.spa")

A_corr = A - B
A_corr.plot()
```

or if all your spectra are in a dataset `A` where the background spectrum is the first one (index `0`):

```python

A = scp.read("mydata/all_spectra.spg")
A_corr = A[1:] - A[0]
A_corr.plot()
```

You can test this directly with one of the sample datasets in spectrochempy. In this case  the reference spectrum is the last one (index `-1`):

```python
import spectrochempy as scp

A = scp.read("irdata/nh4y-activation.spg")
A.plot()

A_corr = A[:-1] - A[-1]
A_corr.plot()
scp.show()
```

Below is the plot of the dataset `A` before correction:
![image](https://github.com/spectrochempy/spectrochempy/assets/10098314/c70765ca-da31-4eb2-958f-38f6438bb97b)

and `A_corr`:
![image](https://github.com/spectrochempy/spectrochempy/assets/10098314/0825a9b7-60ae-49f4-82ed-e27f4acc3b84)

I hope this helps.
Best regards
Arnaud
